src/compositor/lipstickcompositor.cpp

```diff
@@ -732,3 +737,35 @@ void LipstickCompositor::readContent()
 {
     m_recorder->recordFrame(this);
 }
+
+void LipstickCompositor::timerEvent(QTimerEvent *e)
+{
+    if (isVisible()) {
+        stopRendering();
+        QDBusConnection::systemBus().send(m_displayOffReply);
+        // trigger frame callbacks which are pending already at this time
+        surfaceCommitted();
+    } else {
+        frameStarted();
+        sendFrameCallbacks(surfaces());
+        m_fakeRepaintTriggered = false;
+    }
+
+    killTimer(e->timerId());
+}
+
+void LipstickCompositor::stopRendering()
+{
+    hide();
```

Comment:
the visiblity signal is sent only at this point, right? Or is there an earlier (another?) signal from mce that triggers client visibility notifications?